src/main/java/io/airlift/slice/Slice.java

```diff
@@ -311,6 +326,15 @@ public long getRetainedSize()
     }
 
     /**
+     * A slice is considered compact if the base object is an array and it contains the whole array.
+     * As a result, it cannot be a view of a bigger slice.
+     */
+    public boolean isCompact()
+    {
+        return compact;
```

Comment:
Rather than using a field, which increases the object size, I think we can compute this:
```java
return (reference == null) && ((base == null) || (size == base.length));